src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PluginSettings.java

```diff
@@ -113,6 +114,11 @@ public void overrideProperty(String key, String value) {
     settings.setProperty(key, value);
   }
 
+  @VisibleForTesting
+  public @Nullable String getProperty(String key) {
```

Comment:
This annotation should go above the method if you're saying that the returned String may be null

Reply:
I think you missed this one, if this is changed I can +1 the diff

Reply:
I missed this one. Pushed the changes again.